packages/strapi/lib/commands/build.js

```diff
@@ -26,10 +26,50 @@ module.exports = async ({ clean, optimization }) => {
     process.exit(1);
   }
 
-  const serverConfig = await loadConfigFile(envConfigDir, 'server.+(js|json)');
+  const conf = await loadConfigFile(envConfigDir, 'server.+(js|json)');
 
-  const adminPath = _.get(serverConfig, 'admin.path', '/admin');
-  const adminBackend = _.get(serverConfig, 'admin.build.backend', '/');
+  // Defines serverUrl value
+  let serverUrl = _.get(conf, 'url', '');
+  serverUrl = _.trim(serverUrl, '/ ');
+  if (typeof serverUrl !== 'string') {
+    console.log('Invalid server url config. Make sure the url is a string.');
+    process.exit(1);
+  }
+  if (serverUrl.startsWith('http')) {
+    try {
+      serverUrl = _.trim(new URL(conf.url).toString(), '/');
```

Comment:
Isn't this line a bit equivalent to the L.33?

Reply:
In addition It verifies it is a valid url (but only if it starts with "http", because it can be just a path)